### PR TITLE
Use phpunit appropriate methods in getMockForModel

### LINKED ISSUE #14747: TestCase::getMockForModel generate phpunit deprecations notices
This is a (multiple allowed):

* [x] bug
* [x] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 4.1-RC2.
* Platform and Target: phpunit 8.5.8, php 7.2

### What you did
Have test cases with model linked with behaviors and tried mock some of behavior methods.

### What happened
Get that message when tests are run:

`Trying to set mock method "myBehaviorMethod" with onlyMethods, but it does not exist in class "App\Model\Table\ArticlesTable". Use addMethods() for methods that don't exist in the class.`

### What you expected to happen
Mocked methods and silent run

I found a PR on PHPUNIT about that deprecation: https://github.com/sebastianbergmann/phpunit/pull/3687

So, inspecting `Cake\TestSuite\TestCase::getMockForModel` I can fix that message changing: 

```php
        $options += $locator->getConfig($alias);

        /** @var \Cake\ORM\Table $mock */
        $mock = $this->getMockBuilder($className)
            ->onlyMethods($methods)
            ->setConstructorArgs([$options])
            ->getMock();
```

to

```php
        $options += $locator->getConfig($alias);
        $existingMethods = array_intersect($methods, get_class_methods($className));
        $nonExistingMethods = array_diff($methods, $existingMethods);

        /** @var \Cake\ORM\Table $mock */
        $mock = $this->getMockBuilder($className)
            ->onlyMethods($existingMethods)
            ->addMethods($nonExistingMethods)
            ->setConstructorArgs([$options])
            ->getMock();
```

### PR DESCRIPTION
**Pull request by CauanCabral**
Fix #14747

### CODE REVIEW

**Comment by dereuromark** [comment]
If this can be done in tests without any bc issues for src/, this can also go into master.

**Comment by CauanCabral** [comment]
Many tests broken know, I'll look with time later

**Comment by CauanCabral** [comment]
Tests fixed, that's more easy as I think.

File: src/TestSuite/TestCase.php [hunk 1]
```diff
@@ -856,12 +856,16 @@ public function getMockForModel(string $alias, array $methods = [], array $optio
...
-        /** @var \Cake\ORM\Table $mock */
-        $mock = $this->getMockBuilder($className)
+        $builder = $this->getMockBuilder($className)
             ->onlyMethods($existingMethods)
-            ->addMethods($nonExistingMethods)
-            ->setConstructorArgs([$options])
-            ->getMock();
+            ->setConstructorArgs([$options]);
+
+        if (!empty($nonExistingMethods)) {
```
Comment (dereuromark) [inline src/TestSuite/TestCase.php:863]: If ($...) 
The cloaking check is not necessary

Comment (CauanCabral) [reply src/TestSuite/TestCase.php:863]: Sorry, I don't understand "the cloaking check", is the `!empty` part?

**Comment by CauanCabral** [comment]
Include a test and rebased

File: tests/test_app/TestApp/Model/Table/SluggedPostsTable.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * @since         4.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace TestApp\Model\Table;
+
+use Cake\ORM\Table;
+
+class SluggedPostsTable extends Table
+{
+    protected $_table = 'posts';
```
Comment (othercorey) [inline tests/test_app/TestApp/Model/Table/SluggedPostsTable.php:24]: We should use the setter in constructor.

**Pull request merged**